chaos_genius/views/kpi_view.py

```diff
@@ -429,8 +430,7 @@ def edit_kpi(kpi_id):
                         "overall": True,
                         "subdim": True,
                     }
-
-                if kpi_obj.anomaly_params is not None and (
+                if (kpi_obj.run_anomaly is True) and (kpi_obj.anomaly_params is not None) and (
```

Comment:
I think this condition is not needed here. We should update `run_optional` even if anomaly isn't running.

Reply:
I don't see how anomaly being disabled affects this. Not doing so leads to a bug though - anomaly will fail if dimensions were removed while anomaly was disabled.

Reply:
Yeah, there can be a case where the user disables anomaly then removes subdim from the edit KPI screen.